src/eduid_userdb/element.py

```diff
@@ -531,17 +546,28 @@ def _get_primary(self, elements):
         """
         if not elements:
             return None
-        res = [x for x in elements if x.is_primary is True]
+        verified = [x for x in elements if x.is_verified is True]
+        if len(verified) == 0:
+            try:
+                assert len([e for e in elements if e.is_primary]) == 0
+            except AssertionError:
+                raise PrimaryElementViolation('There are unconfirmed primary elements')
+            return None
+        res = [x for x in verified if x.is_primary is True]
```

Comment:
The idea was to replace all uses of <elem-list>.primary.key with <elem-list>.verified.primary.key where needed. And I think as you suggest, that using verified() would create an infinite loop. 

I do realize now that I should be using self.__class__(...) rather than PrimaryElementList(...), to get NinLists etc.